Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm

```diff
@@ -1163,6 +1164,17 @@ static bool exceedsRenderTreeSizeSizeThreshold(uint64_t thresholdSize, uint64_t
 
 #endif
 
+void WebPageProxy::platformDidCommitLoadForFrame(const WebFrameProxy& frame)
+{
+    if (!frame.isMainFrame())
+        return;
+
+    if (!preferences().textExtractionEnabled())
+        return;
+
+    prepareTextExtractionSupportIfNeeded();
```

Comment:
Especially, the way this is written (using an early return), doesn't encourage further use of this method for other platform-specific logic.

Reply:
Moved this logic inline into `didCommitLoadForFrame` — thanks for the review!